### PR TITLE
Fpay 15 completion payment

### PR DESCRIPTION
**Pull request by MichaelYoung1981**

### CODE REVIEW

File: src/Domain.UnitTests/ApprenticeshipFunding/WhenCalculateEarnings.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using AutoFixture;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace SFA.DAS.Funding.ApprenticeshipEarnings.Domain.UnitTests.ApprenticeshipFunding
+{
+    [TestFixture]
+    public class WhenCalculateEarnings
```
Comment (ruslanss) [inline src/Domain.UnitTests/ApprenticeshipFunding/WhenCalculateEarnings.cs:9]: is this WIP? 😁

File: src/Domain.UnitTests/Factories/ApprenticeshipFactory/WhenCreatingANewApprenticeship.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            Assert.AreEqual(apprenticeshipEntityModel.EmployerAccountId, apprenticeship.EmployerAccountId);
+            Assert.AreEqual(apprenticeshipEntityModel.FundingEmployerAccountId, apprenticeship.FundingEmployerAccountId); 
+            Assert.AreEqual(apprenticeshipEntityModel.LegalEntityName, apprenticeship.LegalEntityName);
+            Assert.AreEqual(apprenticeshipEntityModel.PlannedEndDate, apprenticeship.PlannedEndDate);
+            Assert.AreEqual(apprenticeshipEntityModel.TrainingCode, apprenticeship.TrainingCode);
+            Assert.AreEqual(apprenticeshipEntityModel.UKPRN, apprenticeship.UKPRN);
+            Assert.AreEqual(apprenticeshipEntityModel.ApprenticeshipKey, apprenticeship.ApprenticeshipKey);
+            Assert.AreEqual(apprenticeshipEntityModel.ApprovalsApprenticeshipId, apprenticeship.ApprovalsApprenticeshipId);
+            Assert.AreEqual(apprenticeshipEntityModel.Uln, apprenticeship.Uln);
+
```
Comment (ruslanss) [inline src/Domain.UnitTests/Factories/ApprenticeshipFactory/WhenCreatingANewApprenticeship.cs:38]: `.ShouldBeEquivalentTo` could do this in one line

File: src/Domain/Apprenticeship/EarningsProfile.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        public EarningsProfile(decimal adjustedPrice, List<Instalment> instalments, decimal completionPayment)
+        {
+            AdjustedPrice = adjustedPrice;
+            Instalments = instalments;
+            CompletionPayment = completionPayment;
+        }
+
+        public decimal AdjustedPrice { get; set; }
+        public List<Instalment> Instalments { get; set; }
+        public decimal CompletionPayment { get; set; }
```
Comment (ruslanss) [inline src/Domain/Apprenticeship/EarningsProfile.cs:14]: should we avoid public setters in domain models?

File: src/Domain/Apprenticeship/Instalment.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        public Instalment(short academicYear, byte deliveryPeriod, decimal amount)
+        {
+            AcademicYear = academicYear;
+            DeliveryPeriod = deliveryPeriod;
+            Amount = amount;
+        }
+
+        public short AcademicYear { get; set; }
+        public byte DeliveryPeriod { get; set; }
+        public decimal Amount { get; set; }
```
Comment (ruslanss) [inline src/Domain/Apprenticeship/Instalment.cs:14]: should we avoid public setters in domain models?

File: src/Domain/ApprenticeshipFunding/Earning.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿namespace SFA.DAS.Funding.ApprenticeshipEarnings.Domain.ApprenticeshipFunding
+{
+    public class Earning
+    {
+        public short AcademicYear { get; set; }
+        public byte DeliveryPeriod { get; set; }
+        public decimal Amount { get; set; }
```
Comment (ruslanss) [inline src/Domain/ApprenticeshipFunding/Earning.cs:7]: should we avoid public setters in domain models?

**Review by ruslanss — DISMISSED** [review]

**Review by tom-gough — APPROVED** [review]

**Pull request merged**